backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/services/EditingContextEPackageService.java

```diff
@@ -0,0 +1,93 @@
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.emf.services;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.emf.ecore.impl.EPackageRegistryImpl;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+import org.eclipse.sirius.emfjson.resource.JsonResource;
+import org.eclipse.sirius.web.domain.Domain;
+import org.eclipse.sirius.web.domain.DomainPackage;
+import org.eclipse.sirius.web.emf.domain.DomainConverter;
+import org.eclipse.sirius.web.persistence.entities.DocumentEntity;
+import org.eclipse.sirius.web.persistence.repositories.IDocumentRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+/**
+ * A service used to retrieve all the EPackages accessible for given editing context.
+ *
+ * @author sbegaudeau
+ */
+@Service
+public class EditingContextEPackageService implements IEditingContextEPackageService {
+
+    private final Logger logger = LoggerFactory.getLogger(EditingContextEPackageService.class);
+
+    private final IDocumentRepository documentRepository;
+
+    public EditingContextEPackageService(IDocumentRepository documentRepository) {
+        this.documentRepository = Objects.requireNonNull(documentRepository);
+    }
+
+    @Override
+    public List<EPackage> getEPackages(UUID editingContextId) {
+        // @formatter:off
+        List<DocumentEntity> entities = StreamSupport.stream(this.documentRepository.findAll().spliterator(), false)
+            .filter(documentEntity -> documentEntity.getName().contains(".domain")) //$NON-NLS-1$
+            .collect(Collectors.toList());
+        // @formatter:on
+
+        EPackageRegistryImpl ePackageRegistry = new EPackageRegistryImpl();
+        ePackageRegistry.put(DomainPackage.eNS_URI, DomainPackage.eINSTANCE);
+        ResourceSet resourceSet = new ResourceSetImpl();
+        resourceSet.setPackageRegistry(ePackageRegistry);
+
+        for (DocumentEntity documentEntity : entities) {
+            URI uri = URI.createURI(documentEntity.getId().toString());
+            JsonResource resource = new SiriusWebJSONResourceFactoryImpl().createResource(uri);
+            try (var inputStream = new ByteArrayInputStream(documentEntity.getContent().getBytes())) {
+                resource.load(inputStream, null);
+
+                resource.eAdapters().add(new DocumentMetadataAdapter(documentEntity.getName()));
+                resourceSet.getResources().add(resource);
```

Comment:
I think `resourceSet.getResources().add(resource);` should be done before the call to `load()`, otherwise `load()` is invoked in a contexte in which setting up `DomainPackage` in the `ResourceSet`'s registry has no effect (simila to my patch [here][https://github.com/eclipse-sirius/sirius-components/pull/358/files#diff-1e6fcee81e8f924a074f8502433c2acc3602eb63fccef5860ac2dd23f175cbe0)